src/JHipsterNetSampleApplication/Program.cs

```diff
@@ -1,17 +1,75 @@
 using Microsoft.AspNetCore;
 using Microsoft.AspNetCore.Hosting;
+using Microsoft.Extensions.Configuration;
+using Serilog;
+using System;
+using System.IO;
+using ILogger = Serilog.ILogger;
 
 namespace JHipsterNetSampleApplication {
     public class Program {
-        public static void Main(string[] args)
+
+        public static int Main(string[] args)
         {
-            CreateWebHostBuilder(args).Build().Run();
+            try {
+
+                ConfigureLogger();
+
+                Log.Information($"Starting web host");
+                CreateWebHostBuilder(args).Build().Run();
+                return 0;
+
+            }
+            catch (Exception ex) {
+
+                Log.Fatal(ex, $"Host terminated unexpectedly");
+                return 1;
+
+            }
+            finally {
+
+                Log.CloseAndFlush();
+
+            }
+
+
         }
 
         public static IWebHostBuilder CreateWebHostBuilder(params string[] args)
         {
             return WebHost.CreateDefaultBuilder(args)
-                .UseStartup<Startup>();
+                .UseStartup<Startup>()
+                .UseSerilog();
+        }
+
+        /// <summary>
+        /// Gets the current application configuration
+        /// from global and specific appsettings.
+        /// </summary>
+        /// <returns>Return the current <see cref="IConfiguration"/></returns>
+        private static IConfiguration GetAppConfiguration()
+        {
+            var environment = Environment.GetEnvironmentVariable("ASPNETCORE_ENVIRONMENT");
```

Comment:
I thought it was ASP.Net Core default behavior ?
Same for the `appsettings.json`, `appsettings.{environment}.json`.
Do we have any benefit to explicit that part rather than relying on the default config.?

Reply:
In main entrypoint, we are not in an Asp.Net context by default but a simple C# Application context. Because of this, we don't have access to configuration to setup Serilog (it's `CreateDefaultBuilder` who add Configuration). 

We must create an `IConfiguration` instance before asp.net bootstrap if we want to use serilog at the hightest possible level.

Your snippet is good, it use a callback to give you access to the asp.net context when It will be setup. But it will prevent your application to catch exceptions with Serilog during the bootstrap process.

My snippet put Serilog at the entrypoint level to catch all possible exceptions. But we need to do extra work at startup.

So it depend if we want to catch startup logs with Serilog or not. We can argue that startup exceptions are special cases and we don't want to give Serilog responsability of those exceptions.

Reply:
The main issue I have with a more "global" serilog bootstrap, is that it doesn't capture the `SourceContext` (since AspNet Core is not yet starter)....

Anyway, the PR is good, let's move on to the configuration of the logger